packages/rum-core/test/common/event-target-patch.spec.js

```diff
@@ -0,0 +1,164 @@
+ * THE SOFTWARE.
+ *
+ */
+
+import patchEventHandler from './patch'
+import { EVENT_TARGET } from '../../src/common/constants'
+
+describe('EventTargetPatch', function() {
+  let events = []
+  let cancelFn
+
+  beforeAll(function() {
+    cancelFn = patchEventHandler.observe(EVENT_TARGET, function(event, task) {
+      events.push({
+        event,
+        task
+      })
+    })
+  })
+
+  afterAll(function() {
+    cancelFn()
+  })
+
+  beforeEach(function() {
+    events = []
+  })
+
+  function createEvent(eventType) {
+    let event
+    if (typeof Event === 'function') {
+      event = new Event(eventType)
+    } else {
+      /**
+       * For IE11 support
+       */
+      event = document.createEvent('Event')
+      event.initEvent(eventType, true, true)
+    }
+    return event
+  }
+
+  it('should work different argument types', () => {
+    let element = document.createElement('div')
+    document.body.insertBefore(element, null)
+    let eventType = 'click'
+    element.addEventListener(eventType, undefined)
+    let spy = jasmine.createSpy('spy')
+    element.addEventListener(eventType, spy)
+    element.click()
+    expect(spy).toHaveBeenCalled()
+  })
+
+  it('should not affect other event types', () => {
+    const eventType = 'apm-test-event'
+    let event = createEvent(eventType)
+
+    let count = 0
+    let element = document.createElement('div')
+    const listener = e => {
+      expect(e.type).toBe(eventType)
+      count++
+    }
+    element.addEventListener(eventType, listener)
+    element.dispatchEvent(event)
+    expect(count).toBe(1)
+    expect(events.length).toBe(0)
+    element.removeEventListener(eventType, listener)
+    element.dispatchEvent(event)
+    expect(count).toBe(1)
+  })
+
+  if (window.EventTarget) {
+    it('should patch EventTarget', () => {
+      let count = 0
+      let element = document.createElement('div')
+      const listener = e => {
+        expect(e.type).toBe('click')
+        count++
+      }
+
+      element.addEventListener('click', listener)
+      element.addEventListener('click', listener, { capture: true })
+      element.click()
+      expect(count).toBe(2)
+      element.removeEventListener('click', listener)
+      element.click()
+      expect(count).toBe(3)
+      element.removeEventListener('click', listener, true)
+      element.click()
+      expect(events.map(e => e.event)).toEqual([
+        'schedule',
+        'invoke',
+        'schedule',
+        'invoke',
+        'schedule',
+        'invoke'
+      ])
+      expect(count).toBe(3)
+    })
+
+    it('should consider event listener uniqueness condition', () => {
+      let count = 0
+      let element = document.createElement('div')
+      const listener = e => {
+        expect(e.type).toBe('click')
+        count++
+      }
+
+      element.addEventListener('click', listener)
+      element.addEventListener('click', listener)
+      /**
+       * Form with options object is only supported on
```

Comment:
I didn't quite get this comment. what do you mean by `form` here? I am confused because this comment says we cannot pass object and line 144 does exactly that.